awscli/customizations/ec2/paginate.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+class EC2PageSizeInjector(object):
+
+    DEFAULT_TARGET_OPERATIONS = {
+        "describe-volumes": [],
+        "describe-snapshots": ['owner-ids', 'restorable_by_user_ids']
+    }
+
+    DEFAULT_GLOBAL_WHITELIST = [
+        'cli_input_json', 'generate_cli_skeleton', 'help', 'max_items',
+        'next_token', 'dry_run', 'starting_token'
+    ]
+
+    def __init__(self, default_page_size=1000, target_operations=None,
```

Comment:
Are the ``target_operations`` and ``global_whitelist`` needed? It is really only used for testing and if that is the case why can't you just keep the class variables, instantiate an instance, and then modify the list or dictionary for the test.

Another reason I am asking is that I find the ``target_operations`` interface to be strange. Strange in that it will be a dictionary. I would have more expected that to be a list of operations to inject into and then you would have you class variable ``PER_OPERATION_WHITELIST`` which will add specific parameters to the ``GLOBAL_WHITELIST`` for that operation if it is present in ``PER_OPERATION_WHITELIST``.